eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionReportWriter.java

```diff
@@ -0,0 +1,185 @@
+
+    private static final String IS_HEADER_WRITTEN_VALUE = "true";   // use string because ExecutionContext doesn't support boolean
+
+    private static final Logger logger = LoggerFactory.getLogger(AccessionReportWriter.class);
+
+    private final File output;
+
+    private FastaSequenceReader fastaSequenceReader;
+
+    private BufferedWriter fileWriter;
+
+    private String accessionPrefix;
+
+    public AccessionReportWriter(File output, FastaSequenceReader fastaSequenceReader) throws IOException {
+        this.fastaSequenceReader = fastaSequenceReader;
+        this.output = output;
+        this.accessionPrefix = "ss";
+    }
+
+    public String getAccessionPrefix() {
+        return accessionPrefix;
+    }
+
+    public void setAccessionPrefix(String accessionPrefix) {
+        this.accessionPrefix = accessionPrefix;
+    }
+
+    @Override
+    public void open(ExecutionContext executionContext) throws ItemStreamException {
+        boolean isHeaderAlreadyWritten = IS_HEADER_WRITTEN_VALUE.equals(executionContext.get(IS_HEADER_WRITTEN_KEY));
+        if (output.exists() && !isHeaderAlreadyWritten) {
+            logger.warn("According to the job's execution context, the accession report should not exist, but it does" +
+                                " exist. The AccessionReportWriter will append to the file, but it's possible that " +
+                                "there will be 2 non-contiguous header sections in the report VCF. This can happen if" +
+                                " the job execution context was not properly retrieved from the job repository.");
+        }
+        try {
+            boolean append = true;
+            this.fileWriter = new BufferedWriter(new FileWriter(this.output, append));
+            if (!isHeaderAlreadyWritten) {
+                writeHeader();
+                executionContext.put(IS_HEADER_WRITTEN_KEY, IS_HEADER_WRITTEN_VALUE);
+            }
+        } catch (IOException e) {
+            throw new ItemStreamException(e);
+        }
+    }
+
+    private void writeHeader() throws IOException {
+        fileWriter.write("##fileformat=VCFv4.2");
+        fileWriter.newLine();
+        fileWriter.write("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO");
+        fileWriter.newLine();
+    }
+
+    @Override
+    public void update(ExecutionContext executionContext) throws ItemStreamException {
+
+    }
+
+    @Override
+    public void close() throws ItemStreamException {
+        try {
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new ItemStreamException(e);
+        }
+    }
+
+    public void write(Map<Long, ISubmittedVariant> accessions) throws IOException {
+        if (fileWriter == null) {
+            throw new IOException("The file " + output + " was not opened properly. Hint: Check that the code " +
+                                          "called AccessionReportWriter::open");
```

Comment:
This message looks incomplete to me:
"Check that the code called AccessionReportWriter::open"

Reply:
```
[you should]
check that (the code called AccessionReportWriter::open)
            ^^^^^^^^ ^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^^^^^
             subject  verb         direct object
```
I don't know how to make the sentence clearer

Reply:
I see! I had not understood the sentence well when I read it.